src/Provider/Yandex/Model/YandexAddress.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\Yandex\Model;
+
+use Geocoder\Model\Address;
+
+/**
+ * @author Tobias Nyholm <tobias.nyholm@gmail.com>
+ */
+class YandexAddress extends Address
+{
+    /**
+     * @var string|null
+     */
+    private $precision;
+
+    /**
+     * The name of this location.
+     *
+     * @var string|null
+     */
+    private $name;
+
+    /**
+     * @return null|string
+     */
+    public function getPrecision()
+    {
+        return $this->precision;
+    }
+
+    /**
+     * @param null|string $precision
+     *
+     * @return YandexAddress
+     */
+    public function setPrecision($precision)
```

Comment:
> Your suggestion means that the (already big and ugly) constructor of Address has to be overridden. Is that somehow avoidable? Could we make the constructor prettier?

OK, it is "big and ugly". What's the problem with it? If `Address` requires a lot of params, it is natural that constructor is "big and ugly". Setter doesn't make code "prettier", it just makes object less explicit, it hides the problem (if it does exist).

If you ask me, many arguments for data object is OK, it is not a service.